avocado/core/job.py

```diff
@@ -275,6 +275,9 @@ def _make_test_suite(self, references=None):
         loader.loader.load_plugins(self.args)
         try:
             suite = loader.loader.discover(references)
```

Comment:
Yep, I agree with all of the points you raised.  Hopefully when we have another filtering use case we can try creating a uniform interface.